src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/CompletionMessage.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
+{
+    public class CompletionMessage : HubMessage
+    {
+        public string Error { get; }
+        public object Result { get; }
+        public bool HasResult { get; }
```

Comment:
If you return `null` from a Hub method, we "have" a result, but it's `null`. Need to distinguish that from a CompletionMessage that has no `result` at all